src/container.cpp

```diff
@@ -582,13 +580,13 @@ void Container::replaceThing(uint32_t index, Thing* thing)
 	replacedItem->setParent(nullptr);
 }
 
-void Container::removeThing(Thing* thing, uint32_t count)
+void Container::removeThing(std::shared_ptr<Thing> thing, uint32_t count)
 {
 	if (!thing) {
 		return;
 	}
 
-	Item* item = thing->getItem();
+	auto item = thing->getItem();
```

Comment:
Could this be `const auto`
```suggestion
	const auto item = thing->getItem();
```